crates/api_crud/src/site/update.rs

```diff
@@ -67,22 +67,27 @@ pub async fn update_site(
     SiteLanguage::update(&mut context.pool(), discussion_languages.clone(), &site).await?;
   }
 
-  replace_image(&data.icon, &site.icon, &context).await?;
-  replace_image(&data.banner, &site.banner, &context).await?;
-
   let slur_regex = local_site_to_slur_regex(&local_site);
   let url_blocklist = get_url_blocklist(&context).await?;
-  let sidebar = process_markdown_opt(&data.sidebar, &slur_regex, &url_blocklist, &context).await?;
-  let icon = proxy_image_link_opt_api(&data.icon, &context).await?;
-  let banner = proxy_image_link_opt_api(&data.banner, &context).await?;
+  let sidebar = diesel_string_update(
+    &process_markdown_opt(&data.sidebar, &slur_regex, &url_blocklist, &context).await?,
+  );
+
+  let icon = diesel_url_update(&data.icon)?;
```

Comment:
The heart of it is here. We transform these url-type ones into proper diesel forms for an update to an optional url column (IE `Option<Option<DbUrl>>` ), before doing later actions to it.